e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/gef/submit-deal-cancellation-effective-from-in-future.spec.js

```diff
@@ -0,0 +1,125 @@
+import { TFM_DEAL_STAGE, TFM_FACILITY_STAGE } from '@ukef/dtfs2-common';
+import relative from '../../../relativeURL';
+import { ADMIN, BANK1_MAKER1, PIM_USER_1, T1_USER_1 } from '../../../../../../e2e-fixtures';
+import caseDealPage from '../../../pages/caseDealPage';
+import { today, tomorrow } from '../../../../../../e2e-fixtures/dateConstants';
+import dealsPage from '../../../pages/dealsPage';
+import { caseSubNavigation, successBanner } from '../../../partials';
+import facilitiesPage from '../../../pages/facilitiesPage';
+import activitiesPage from '../../../pages/activities/activitiesPage';
+import { MOCK_APPLICATION_AIN } from '../../../../fixtures/mock-gef-deals';
+import { anIssuedCashFacility } from '../../../../../../e2e-fixtures/mock-gef-facilities';
+import { DEAL_TYPE } from '../../../../../../gef/cypress/fixtures/constants';
+
+context('Deal cancellation - submit cancellation with "effectiveFrom" in future', () => {
+  let dealId;
+  let facility;
+  const ukefDealId = 10000001;
+
+  before(() => {
+    cy.insertOneGefDeal(MOCK_APPLICATION_AIN, BANK1_MAKER1).then((insertedDeal) => {
+      dealId = insertedDeal._id;
+      // updates a gef deal so has relevant fields
+      cy.updateGefDeal(dealId, MOCK_APPLICATION_AIN, BANK1_MAKER1);
+
+      cy.createGefFacilities(dealId, [anIssuedCashFacility()], BANK1_MAKER1).then((createdFacility) => {
+        facility = createdFacility.details;
+      });
+
+      cy.submitDeal(dealId, DEAL_TYPE.GEF, T1_USER_1);
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+    cy.visit(relative(`/case/${dealId}/deal`));
+  });
+
+  after(() => {
+    cy.deleteDeals(dealId, ADMIN);
+    cy.deleteFacility(facility._id, BANK1_MAKER1);
+  });
+
+  describe('when logged in as a PIM user', () => {
+    before(() => {
+      cy.login(PIM_USER_1);
+      cy.visit(relative(`/case/${dealId}/deal`));
+      cy.submitDealCancellation({ dealId, effectiveDate: tomorrow.date });
+    });
+
+    describe('after submitting the cancellation', () => {
+      const expectedSuccessBannerText = `Deal ${ukefDealId} scheduled for cancellation on ${tomorrow.d_MMMM_yyyy}`;
+
+      it('should redirect you to the deal summary page', () => {
+        cy.url().should('eq', relative(`/case/${dealId}/deal`));
+      });
+
+      it('should display the deal cancelled banner which should persist on page reload', () => {
+        cy.url().should('eq', relative(`/case/${dealId}/deal`));
+
+        successBanner().should('exist');
+        cy.assertText(successBanner(), expectedSuccessBannerText);
```

Comment:
No need for `.should('exist');` assertions if there is a proceeding `cy.assertText()` assertion, if the element doesn't exist, `assertText` will error that the element cannot be found 😎

Reply:
I think fine to have both - It's a pattern elsewhere, and it's helpful to know if e.g. it exists but the text is just wrong - so can help with debugging

Reply:
I don't think we need these as the assertText will throw the same kind of "element not found" error with IMO a more descriptive error

Not a blocker as this is in other places, marking as resolved